packages/cloud_firestore/ios/Classes/CloudFirestorePlugin.m

```diff
@@ -191,6 +193,45 @@ - (void)handleMethodCall:(FlutterMethodCall *)call result:(FlutterResult)result
     [[_listeners objectForKey:handle] remove];
     [_listeners removeObjectForKey:handle];
     result(nil);
+  } else if ([@"Batch#create" isEqualToString:call.method]) {
+    __block NSNumber *handle = [NSNumber numberWithInt:_nextBatchHandle++];
+    FIRWriteBatch *batch = [[FIRFirestore firestore] batch];
+    _batches[handle] = batch;
+    result(handle);
+  } else if ([@"Batch#set" isEqualToString:call.method]) {
+    NSNumber *handle = call.arguments[@"handle"];
+    NSString *path = call.arguments[@"path"];
+    NSDictionary *options = call.arguments[@"options"];
+    FIRDocumentReference *reference = [[FIRFirestore firestore] documentWithPath:path];
+    FIRWriteBatch batch = [_batches objectForKey:handle];
+    if (![options isEqual:[NSNull null]] &&
+        [options[@"merge"] isEqual:[NSNumber numberWithBool:YES]]){
+      [batch setData:call.arguments[@"data"]
+         forDocument:reference
+             options:[FIRSetOptions merge]];
+    } else {
+      [batch setData:call.arguments[@"data"] forDocument:reference];
+    }
+    result(nil);
+  } else if ([@"Batch#update" isEqualToString:call.method]) {
+    NSNumber *handle = call.arguments[@"handle"];
+    NSString *path = call.arguments[@"path"];
+    FIRDocumentReference *reference = [[FIRFirestore firestore] documentWithPath:path];
+    FIRWriteBatch batch = [_batches objectForKey:handle];
+    [batch updateData:call.arguments[@"data"] forDocument:reference];
+    result(nil);
+  } else if ([@"Batch#delete" isEqualToString:call.method]) {
+    NSNumber *handle = call.arguments[@"handle"];
+    NSString *path = call.arguments[@"path"];
+    FIRDocumentReference *reference = [[FIRFirestore firestore] documentWithPath:path];
+    FIRWriteBatch batch = [_batches objectForKey:handle];
+    [batch deleteDocument:reference];
+    result(nil);
+  } else if ([@"Batch#commit" isEqualToString:call.method]) {
+    // TODO:
```

Comment:
Did you mean to leave this empty TODO?

I think you should remove the batches from the NSDictionary once they are committed